### PR TITLE
refactor: 💡 Migrate `form.input` to `Hds::TextInput::Field`

### PR DESCRIPTION
**Pull request by dvonanderson**
✅ Closes: https://hashicorp.atlassian.net/browse/ICU-12459
✅ Closes: https://hashicorp.atlassian.net/browse/ICU-12460

:tickets: [Jira ticket](https://hashicorp.atlassian.net/browse/ICU-12459)
:tickets: [Jira ticket](https://hashicorp.atlassian.net/browse/ICU-12460)

## Description
- Migrate the use of `form.input` to `Hds::TextInput::Field` for OIDC and Password auth-method form

#### Before and after `Password` auth method form:
---
image
image

#### Before and after `OIDC` auth method form:
---
image
image

## How to Test
- Click on Vercel link for Admin UI or pull down PR and run it locally using yarn start
- Create both a `OIDC` and `Password` auth-method and ensure that all fields changed in this PR get saved 
- Click on the newly created auth-methods from the list and ensure that the data persisted.

:technologist: [Admin preview](PATH_TO_FEATURE)

## Checklist:

- [x] I have added before and after screenshots for UI changes
~~- [ ] I have added JSON response output for API changes~~
~~- [ ] I have added steps to reproduce and test for bug fixes in the description~~
~~- [ ] I have commented on my code, particularly in hard-to-understand areas~~
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: ui/admin/app/components/form/auth-method/oidc/index.hbs [hunk 1]
```diff
@@ -48,33 +46,26 @@
...
+  <Hds::Form::TextInput::Field
     @value={{@model.issuer}}
-    @label={{t 'form.issuer.label'}}
-    @helperText={{t 'form.issuer.help'}}
-    @error={{@model.errors.issuer}}
-    readonly={{false}}
-    as |field|
+    @isInvalid={{@model.errors.issuer}}
+    disabled={{form.disabled}}
+    name='issuer'
```
Comment (calcaide) [inline ui/admin/app/components/form/auth-method/oidc/index.hbs:55]: None blocking comment: as in previous implementation, I will suggest keep the attribute `name` the first attribute to facilitate Developer experience (facilitates reading).

**Review by calcaide — APPROVED** [review]
LGTM!! Great HDS adoption!!

**Review by cameronperera — APPROVED** [review]

File: ui/admin/app/components/form/auth-method/password/index.hbs [hunk 2]
```diff
@@ -10,49 +10,40 @@
...
     @value={{@model.description}}
     @isInvalid={{@model.errors.description}}
-    name='description'
     disabled={{form.disabled}}
+    name='description'
     as |F|
   >
     <F.Label>{{t 'form.description.label'}}</F.Label>
     <F.HelperText>{{t 'form.description.help'}}</F.HelperText>
-    {{#if @model.errors.description}}
```
Comment (ZedLi) [inline ui/admin/app/components/form/auth-method/password/index.hbs:49]: Why was this removed? Was it confirmed we don't get errors back for description?

File: ui/admin/app/components/form/auth-method/oidc/index.hbs [hunk 3]
```diff
@@ -406,45 +397,45 @@
...
-    @type='number'
+  <Hds::Form::TextInput::Field
     @value={{@model.max_age}}
-    @label={{t 'form.max_age.label'}}
-    @helperText={{t 'form.max_age.help'}}
-    @error={{@model.errors.max_age}}
-    readonly={{false}}
+    @isInvalid={{@model.errors.max_age}}
     step='1'
     min='0'
```
Comment (ZedLi) [inline ui/admin/app/components/form/auth-method/oidc/index.hbs:404]: Step and min are only valid for number input types

**Review by ZedLi — CHANGES_REQUESTED** [review]

File: ui/admin/app/components/form/auth-method/oidc/index.hbs (hunk 1 above)
Comment (dvonanderson) [reply ui/admin/app/components/form/auth-method/oidc/index.hbs:55]: @calcaide good point

File: ui/admin/app/components/form/auth-method/oidc/index.hbs (hunk 3 above)
Comment (dvonanderson) [reply ui/admin/app/components/form/auth-method/oidc/index.hbs:405]: @ZedLi good catch incoming change

File: ui/admin/app/components/form/auth-method/password/index.hbs (hunk 2 above)
Comment (dvonanderson) [reply ui/admin/app/components/form/auth-method/password/index.hbs:49]: @ZedLi definitely a mishap on my part, but I did looking at the `auth-method` [service](https://github.com/hashicorp/boundary/blob/e95eda7ca2a36434cf83da1454ee7cb49cdee1e6/internal/proto/controller/api/services/v1/auth_method_service.proto#L40) it looks like no errors are returned for description in the request.  Is that you you see it here?

File: ui/admin/app/components/form/auth-method/oidc/index.hbs (hunk 1 above)
Comment (dvonanderson) [reply ui/admin/app/components/form/auth-method/oidc/index.hbs:62]: @calcaide resolve these in last commits

File: ui/admin/app/components/form/auth-method/oidc/index.hbs (hunk 3 above)
Comment (ZedLi) [reply ui/admin/app/components/form/auth-method/oidc/index.hbs]: I don't think the change you made changed anything? It's still setting them

File: ui/admin/app/components/form/auth-method/oidc/index.hbs [hunk 4]
```diff
@@ -95,52 +93,52 @@
...
-      </field.errors>
+      </F.Error>
     {{/if}}
-  </form.input>
+  </Hds::Form::TextInput::Field>
 
-  <form.input
-    @name='client_secret'
-    @type='password'
+  <Hds::Form::TextInput::Field
```
Comment (DhariniJeeva) [inline ui/admin/app/components/form/auth-method/oidc/index.hbs:115]: since the type attr is removed we can now see the password

File: ui/admin/app/components/form/auth-method/oidc/index.hbs [hunk 5]
```diff
@@ -406,45 +404,43 @@
 
   </Form::Field::ListWrapper>
 
-  <form.input
-    @name='max_age'
-    @type='number'
+  <Hds::Form::TextInput::Field
+    name='max_age'
```
Comment (DhariniJeeva) [inline ui/admin/app/components/form/auth-method/oidc/index.hbs:408]: We should not remove the type here, too. Can we add the number type back in?

File: ui/admin/app/components/form/auth-method/password/index.hbs [hunk 6]
```diff
@@ -10,49 +10,40 @@
   @showEditToggle={{if @model.isNew false true}}
   as |form|
 >
-  <form.input
-    @value={{@model.type}}
-    @label={{t 'form.type.label'}}
-    readonly={{true}}
-    @disabled={{true}}
-  />
+  <Hds::Form::TextInput::Field @value={{@model.type}} disabled={{true}} as |F|>
```
Comment (DhariniJeeva) [inline ui/admin/app/components/form/auth-method/password/index.hbs:13]: One extra improvement here we could do is use the `infofield` for read-only types, similar to some other resources such as host set, host, etc.

File: ui/admin/app/components/form/auth-method/oidc/index.hbs [hunk 7]
```diff
@@ -499,13 +495,14 @@
...
   {{#unless @model.isNew}}
-    <form.input
+    <Hds::Form::TextInput::Field
       @value={{@model.callback_url}}
-      @label={{t 'form.callback_url.label'}}
-      @helperText={{t 'form.callback_url.help'}}
-      readonly={{true}}
-      @disabled={{true}}
-    />
+      disabled={{true}}
```
Comment (DhariniJeeva) [inline ui/admin/app/components/form/auth-method/oidc/index.hbs:500]: can we add `isoptional` attr here

File: ui/admin/app/components/form/auth-method/oidc/index.hbs [hunk 8]
```diff
@@ -57,24 +55,24 @@
     {{/if}}
   </Hds::Form::Textarea::Field>
 
-  <form.input
-    @name='issuer'
-    @type='text'
+  <Hds::Form::TextInput::Field
```
Comment (DhariniJeeva) [inline ui/admin/app/components/form/auth-method/oidc/index.hbs:58]: can we add `isoptional` attr here?

File: ui/admin/app/components/form/auth-method/oidc/index.hbs [hunk 9]
```diff
@@ -95,52 +93,52 @@
     </F.HelperText>
   </Hds::Form::Checkbox::Field>
 
-  <form.input
-    @name='client_id'
-    @type='text'
+  <Hds::Form::TextInput::Field
```
Comment (DhariniJeeva) [inline ui/admin/app/components/form/auth-method/oidc/index.hbs:96]: can we add isoptional attr here?

File: ui/admin/app/components/form/auth-method/oidc/index.hbs (hunk 4 above)
Comment (DhariniJeeva) [reply ui/admin/app/components/form/auth-method/oidc/index.hbs:115]: can we add isoptional attr here?

**Review by DhariniJeeva — DISMISSED** [review]
Can we add `isOptional` for all fields that we migrated here?

File: ui/admin/app/components/form/auth-method/oidc/index.hbs (hunk 4 above)
Comment (dvonanderson) [reply ui/admin/app/components/form/auth-method/oidc/index.hbs:115]: @DhariniJeeva good catch yes I will add `@isOptional` argument for those three inputs as well as making this `type` password.

File: ui/admin/app/components/form/auth-method/oidc/index.hbs (hunk 5 above)
Comment (dvonanderson) [reply ui/admin/app/components/form/auth-method/oidc/index.hbs:408]: @DhariniJeeva `type="number"` is not supported in `Hds::TextInput::Field`.  The [documentation](https://helios.hashicorp.design/components/form/text-input?tab=code#formtextinputfield) states it causes accessibility issues.  Even if I try and use `type="number"` here it still comes back as a `type="text"`.

File: ui/admin/app/components/form/auth-method/password/index.hbs (hunk 6 above)
Comment (dvonanderson) [reply ui/admin/app/components/form/auth-method/password/index.hbs:13]: @DhariniJeeva ok yea I saw that in another instance.  Coming in next commit.

File: ui/admin/app/components/form/auth-method/oidc/index.hbs (hunk 3 above)
Comment (dvonanderson) [reply ui/admin/app/components/form/auth-method/oidc/index.hbs]: @ZedLi fixed in last commit

File: ui/admin/app/components/form/auth-method/oidc/index.hbs (hunk 4 above)
Comment (dvonanderson) [reply ui/admin/app/components/form/auth-method/oidc/index.hbs:117]: @DhariniJeeva `type="password"` added back in last commit.  Thank you for the catch

File: ui/admin/app/components/form/auth-method/oidc/index.hbs (hunk 7 above)
Comment (dvonanderson) [reply ui/admin/app/components/form/auth-method/oidc/index.hbs:504]: @DhariniJeeva `@isOptional` added back in last commit.  Thank you for the catch

File: ui/admin/app/components/form/auth-method/oidc/index.hbs (hunk 8 above)
Comment (dvonanderson) [reply ui/admin/app/components/form/auth-method/oidc/index.hbs:58]: @DhariniJeeva `@isOptional` added back in last commit.  Thank you for the catch

File: ui/admin/app/components/form/auth-method/oidc/index.hbs (hunk 9 above)
Comment (dvonanderson) [reply ui/admin/app/components/form/auth-method/oidc/index.hbs:97]: @DhariniJeeva `@isOptional` added back in last commit.  Thank you for the catch

File: ui/admin/app/components/form/auth-method/oidc/index.hbs [hunk 10]
```diff
@@ -95,52 +94,55 @@
...
-    @label={{t 'form.client_secret.label'}}
-    @helperText={{t 'form.client_secret.help'}}
-    @error={{@model.errors.client_secret}}
-    readonly={{false}}
-    @hasVisibilityToggle={{false}}
-    as |field|
+    @isInvalid={{@model.errors.client_secret}}
+    @isOptional={{true}}
+    disabled={{form.disabled}}
+    type='password'
```
Comment (ZedLi) [inline ui/admin/app/components/form/auth-method/oidc/index.hbs:123]: I think we still need the `@hasVisibilityToggle={{false}}`

Comment (dvonanderson) [reply ui/admin/app/components/form/auth-method/oidc/index.hbs:124]: @ZedLi ahh good catch, fixed

File: ui/admin/app/components/form/auth-method/oidc/index.hbs (hunk 4 above)
Comment (ZedLi) [reply ui/admin/app/components/form/auth-method/oidc/index.hbs:117]: This should have been `@type` and not `type`
